app/invocation/scorecard_card.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import InvocationModel from "./invocation_model";
+import rpcService from "../service/rpc_service";
+
+interface Props {
+  model: InvocationModel;
+}
+
+export default class ScorecardCardComponent extends React.Component {
+  props: Props;
+
+  getCacheAddress() {
+    let address = this.props.model.optionsMap.get("remote_cache").replace("grpc://", "");
+    address = address.replace("grpcs://", "");
+
+    if (this.props.model.optionsMap.get("remote_instance_name")) {
+      address = address + "/" + this.props.model.optionsMap.get("remote_instance_name");
+    }
+    return address;
+  }
+
+  getActionDownloadURL(actionId: string) {
+    let actionResultURI = "actioncache://" + this.getCacheAddress() + "/blobs/ac/" + actionId + "/141";
+    return rpcService.getBytestreamFileUrl(actionId, actionResultURI, this.props.model.getId());
+  }
+
+  render() {
+    return (
```

Comment:
might need a guard clause like the following, to avoid an NPE if the invocation doesn't have a scorecard:

```
if (!this.props.model.scoreCard) return null;
```

Reply:
Done.